### PR TITLE
Papp 741 add support for batch operations in unit of work pattern

### PR DESCRIPTION
**Pull request by Maxim-Zavidei**
# Overview:
* [PAPP-741] Added support for batch operations in Unit Of Work pattern
* [PAPP-741] Added disposable functionality to UoW
* [PAPP-741] Decoupled UoW from every Data Access Object
* [PAPP-741] Defined a Factory pattern for generic data access objects implemented by UoW
* [PAPP-741] Further decoupled UoW from the PappDbContext, it now works with any DbContext class,
(This will be useful if in the future we decide to have schemas in our database, or when we will have many backends and some might need to use different contexes, or if we implemnt a login system with tables separate from pappcontext)
* [PAPP-741] Moved the context to a dedicated folder
* [PAPP-741] Updated the tests
* [PAPP-741] Tried to write docs on most the UoW pattern

[PAPP-741]: https://papp.atlassian.net/browse/PAPP-741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-741]: https://papp.atlassian.net/browse/PAPP-741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-741]: https://papp.atlassian.net/browse/PAPP-741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-741]: https://papp.atlassian.net/browse/PAPP-741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-741]: https://papp.atlassian.net/browse/PAPP-741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-741]: https://papp.atlassian.net/browse/PAPP-741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-741]: https://papp.atlassian.net/browse/PAPP-741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-741]: https://papp.atlassian.net/browse/PAPP-741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Papp.Persistence/Shared/UnitOfWork.cs [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            this.Logger.LogError("[UnitOfWork]: The ongoing transaction hasn't yet been commited.");
+        }
+    }
+
+    /// <inheritdoc/>
+    public void Commit()
+    {
+        if (this.Transaction != null)
+        {
+            this.Transaction.Commit();
```
Comment (Logstor) [inline Papp.Persistence/Shared/UnitOfWork.cs:64]: When you commit, shouldn't the `Transaction` not be set to null, so you can start another one with the same UnitOfWork class? :slightly_smiling_face:

File: Papp.Persistence/Shared/UnitOfWork.cs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            this.Logger.LogError("[UnitOfWork]: No ongoing transaction has been started.");
+        }
+    }
+
+    /// <inheritdoc/>
+    public async Task CommitAsync()
+    {
+        if (this.Transaction != null)
+        {
+            await this.Transaction.CommitAsync();
```
Comment (Logstor) [inline Papp.Persistence/Shared/UnitOfWork.cs:78]: Same here with the `Transaction` :slightly_smiling_face:

**Review by Logstor — APPROVED** [review]
Looks nice, but I just have one things that seems incorrect :slightly_smiling_face: 

If it's intentional, then you can Squash or Merge :+1:

File: Papp.Persistence/Shared/UnitOfWork.cs (hunk 1 above)
Comment (Maxim-Zavidei) [reply Papp.Persistence/Shared/UnitOfWork.cs:64]: Oh yes, that's correct, missed it
Really good catch here 💪

File: Papp.Persistence/Shared/UnitOfWork.cs (hunk 2 above)
Comment (Maxim-Zavidei) [reply Papp.Persistence/Shared/UnitOfWork.cs:78]: 👍

**Pull request merged**